### PR TITLE
Add Header component

### PR DESCRIPTION
**Pull request by im36-123**
Add Header component for SmartHR.

### CODE REVIEW

File: src/themes/createPalette.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ export interface PaletteProperty {
   Warning?: string
   Scrim?: string
   Overlay?: string
+  Hanica_Green?: string
```
Comment (nabeliwo) [inline src/themes/createPalette.ts]: Hanica 🙄

File: src/components/Header/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      </HeaderButton>
+
+      <HeaderDropDown
+        key="headerDropDown-0"
+        dropDownKey="manage"
+        icon="fa-users"
+        menus={props.employeeButtonMenu}
+      >
+        従業員管理
+      </HeaderDropDown>
```
Comment (nabeliwo) [inline src/components/Header/Header.tsx]: Abstract menu content 🙏

Comment (im36-123) [reply src/components/Header/Header.tsx]: If this component is abstracted, the function of Header may not be the same in each product.
So I stopped abstracting this component and hardcoded it.

File: src/components/Header/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+import { HeaderUserDropDown, HeaderUserDropDownProps } from './HeaderUserDropDown'
+import { HeaderEmployeeDropDown, HeaderEmployeeDropDownProps } from './HeaderEmployeeDropDown'
+
+interface Props {
+  logoUrl?: string
+  employeeListButtonLink?: string
+  isAdmin?: boolean
+  headerNotification: HeaderNotificationProps
+  headerEmployeeDropDown: HeaderEmployeeDropDownProps
+  headerUserDropDown: HeaderUserDropDownProps
```
Comment (nabeliwo) [inline src/components/Header/Header.tsx]: [IMO] Is there a need to prefix `Header component` props with `header`?
headerNotification, headerEmployeeDropDown, headerUserDropDown...

File: src/components/Header/HeaderNotification.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  ...props
+}) => (
+  <Wrapper theme={theme}>
+    <Anchor
+      theme={theme}
+      number={props.number}
+      href={props.url ? props.url : '/notifications'}
+      aria-label="通知履歴"
+    >
+      {props.number && props.number > 10 ? '9+' : props.number}
```
Comment (nabeliwo) [inline src/components/Header/HeaderNotification.tsx]: [ask] When props.number is 10, 10 is displayed, and when it is 11, '9+' is displayed. Is this correct?

File: src/components/Header/HeaderButton.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    {props.icon && (
+      <HeaderButtonIcon theme={theme} role="presentation">
+        <Icon name={props.icon}></Icon>
+      </HeaderButtonIcon>
+    )}
+    {props.children && props.children}
+  </Wrapper>
+)
+
+export const HeaderButton = withTheme(HeaderButtonComponent)
```
Comment (nabeliwo) [inline src/components/Header/HeaderButton.tsx]: [MUST] Is HeaderButton used for anything other than Header? If not, it should be included in the Header.
I want the components to be placed directly under 'components' only for user use.

File: src/components/Header/HeaderButton.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  url?: string
+  target?: string
+  icon?: IconProps['name']
+}
+
+const HeaderButtonComponent: React.FC<HeaderButtonProps & InjectedProps> = ({
+  theme,
+  ...props
+}) => (
+  <Wrapper theme={theme} href={props.url} target={props.target && props.target}>
```
Comment (nabeliwo) [inline src/components/Header/HeaderButton.tsx]: [MUST] If props.target is missing, 'undefined' will be entered in the target attribute

File: src/components/Header/HeaderButton.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+export const HeaderButton = withTheme(HeaderButtonComponent)
+
+const Wrapper: any = styled.a`
+  ${({ theme }: InjectedProps) => {
+    return css`
+      display: block;
+      margin: 0;
+      padding: 0 ${theme.size.pxToRem(10)};
+      color: #ffffff;
```
Comment (nabeliwo) [inline src/components/Header/HeaderButton.tsx]: [nits] I want to omit as much as possible '#ffffff' -> '#fff'

File: src/components/Header/HeaderEmployeeDropDown.tsx [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+const ButtonWrapper = styled.button`
+  ${({ theme }: InjectedProps) => {
+    return css`
+      display: flex;
+      align-items: center;
+      margin: 0;
+      padding: 0 ${theme.size.pxToRem(10)};
+      border: none;
+      background: initial;
+      color: #fff;
```
Comment (nabeliwo) [inline src/components/Header/HeaderEmployeeDropDown.tsx]: 🙄

File: src/components/Header/HeaderEmployeeDropDown.tsx [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+`
+const DropdownContent = styled(DropdownContentComponent)`
+  transition: none;
+`
+const MenuList = styled.div`
+  ${({ theme }: InjectedProps) => {
+    return css`
+      border: 1px solid ${theme.palette.BORDER};
+      border-radius: ${theme.size.pxToRem(3)};
+      background-color: #ffffff;
```
Comment (nabeliwo) [inline src/components/Header/HeaderEmployeeDropDown.tsx]: 🙄

File: src/components/Header/HeaderNotification.tsx (hunk 4 above)
Comment (im36-123) [reply src/components/Header/HeaderNotification.tsx]: That’s incorrectly behavior. So I will fix that 💪

File: src/components/Header/HeaderEmployeeDropDown.tsx (hunk 8 above)
Comment (im36-123) [reply src/components/Header/HeaderEmployeeDropDown.tsx]: 😳

File: src/components/Header/HeaderEmployeeDropDown.tsx (hunk 9 above)
Comment (im36-123) [reply src/components/Header/HeaderEmployeeDropDown.tsx]: 😳

File: src/components/Header/HeaderButton.tsx (hunk 5 above)
Comment (im36-123) [reply src/components/Header/HeaderButton.tsx]: HeaderButton component already moved to Header directory. 👍

**Comment by im36-123** [comment]
@nabeliwo 
Thank you fore your feedback. 
I fixed my code, please review again

**Pull request merged**